### PR TITLE
Prevent form data from being consumed by Flask

### PR DESCRIPTION
**Pull request by sohtsuka**
Event data unexpectedly becomes empty when invoking an API request with content types
`multipart/form-data`, `application/x-www-form-urlencoded` or `application-x-url-encoded`.

```
$ cat template.yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Resources:
    HelloWorldFunction:
        Type: AWS::Serverless::Function
        Properties:
            CodeUri: hello_world/build/
            Handler: app.lambda_handler
            Runtime: python2.7
            Events:
                HelloWorld:
                    Type: Api
                    Properties:
                        Path: /hello
                        Method: post

$ cat hello_world/build/app.py
import json

def lambda_handler(event, context):
    return {
        "statusCode": 200,
        "body": event['body']
    }

$ # sam local start-api

$ cat test.txt
TEST

$ curl -d @test.txt http://localhost:3000/hello
no data 
```

The cause is that Flask request's `data` property does not contain the data comes with these content types. `get_data()` method always returns raw data and seems to be more relevant.

After the changes:

```
$ # samdev local start-api

$ curl -d @test.txt http://localhost:3000/hello
TEST
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by billyshambrook** [comment]
This is needed to test slack commands.

**Comment by sanathkr** [comment]
@sohtsuka Any chance you can add the functional & integration tests that @jfuss requested? These tests will greatly improve the value of your change by ensuring this behavior is never broken in future :)

**Comment by sohtsuka** [comment]
@jfuss @sanathkr Thanks for reviewing this. I'll try to add the tests.

File: tests/integration/local/start_api/test_start_api.py [hunk 1]
```diff
@@ -369,6 +369,18 @@ def test_binary_request(self):
         self.assertEquals(response.headers.get("Content-Type"), "image/gif")
         self.assertEquals(response.content, input_data)
 
+    def test_request_with_form_data(self):
+        """
+        Form-encoded data should be put into the Event to Lambda
+        """
+        response = requests.post(self.url + "/echobase64eventbody",
```
Comment (jfuss) [inline tests/integration/local/start_api/test_start_api.py]: I know this endpoint echoes the Event back but the path is specific to base64. For this case it wont affect the outcome but to keep things clean and readable, can you create a new path on the [EchoEventFunction](https://github.com/awslabs/aws-sam-cli/blob/develop/tests/integration/testdata/start_api/template.yaml#L37) and update the path here.

**Comment by sanathkr** [comment]
@sohtsuka I noticed this was your first PR. Congrats 🎉. Merged!

**Pull request merged**